### PR TITLE
feat: use job-processor

### PR DESCRIPTION
**Pull request by moroine**

### CODE REVIEW

**Comment by moroine** [comment]
🚀

File: server/src/commands.ts [hunk 1]
```diff
@@ -34,16 +34,10 @@ program
 
 async function startProcessor(signal: AbortSignal) {
   logger.info(`Process jobs queue - start`);
-  if (config.env !== "local") {
-    await addJob({
-      name: "crons:init",
-      queued: true,
-    });
-  }
```
Comment (moroine) [inline server/src/commands.ts:42]: This is now is `jobs.ts`

File: server/src/common/utils/mongodbUtils.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export const getDatabase = () => {
   return ensureInitialization().db();
 };
 
-export const getDbCollection = <K extends CollectionName>(name: K): Collection<WithoutId<IDocumentMap[K]>> => {
```
Comment (moroine) [inline server/src/common/utils/mongodbUtils.ts:57]: `WithoutId` doesn't work with discriminatedUnion

File: server/src/db/migrations/20231114155041-job_processor.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      update.job_status = uploadDocumentWitJob.import_progress === 100 ? "done" : "error";
+      update.job_error = uploadDocumentWitJob.import_progress === 100 ? null : "Unknown";
+    }
+
+    await db.collection("documents").updateOne({ _id: uploadDocumentWitJob._id }, { $set: update });
+  }
+
+  const mailingListDocumentWittJobAndMailingListCursor = db.collection("documents").aggregate([
+    { $match: { nom_fichier: /^(mailing-list)/, kind: null } },
+    { $addFields: { id: { $toString: "$_id" } } },
```
Comment (moroine) [inline server/src/db/migrations/20231114155041-job_processor.ts:54]: We should consider converting `document_id` to `ObjectId` to be able to lookup directly

Comment (moroine) [reply server/src/db/migrations/20231114155041-job_processor.ts:54]: And more generally all id references

File: server/src/modules/actions/documentContent.actions.ts [hunk 4]
```diff
@@ -7,13 +7,14 @@ type TCreateDocumentContent = Omit<IDocumentContent, "_id" | "updated_at" | "cre
...
+  const documentContent: IDocumentContent = {
+    _id: new ObjectId(),
     ...data,
     updated_at: now,
     created_at: now,
-  });
+  };
 
-  const documentContent = await findOneDocumentContent({ _id });
+  const { insertedId: _id } = await getDbCollection("documentContents").insertOne(documentContent);
```
Comment (moroine) [inline server/src/modules/actions/documentContent.actions.ts:17]: Consequence of removing the `WithoutId` but this is fine to use `new ObjectId()`

**Comment by moroine** [comment]
🚀

**Comment by moroine** [comment]
🚀

**Comment by moroine** [comment]
🚀

**Comment by moroine** [comment]
🚀

**Comment by moroine** [comment]
🚀

**Comment by moroine** [comment]
🚀

File: server/src/db/migrations/20231114155041-job_processor.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
+import { Db, MongoClient } from "mongodb";
+import { IMailingListDocument, IUploadDocument } from "shared/models/document.model";
+
+export const up = async (db: Db, _client: MongoClient) => {
+  const uploadDocumentWitJobCursor = db.collection("documents").aggregate([
```
Comment (david-nathanael) [inline server/src/db/migrations/20231114155041-job_processor.ts:5]:
```suggestion
  const uploadDocumentWithJobCursor = db.collection("documents").aggregate([
```

File: server/src/modules/actions/documents.actions.ts [hunk 6]
```diff
@@ -326,34 +320,14 @@ export const uploadFile = async (
...
-      chemin_fichier: path,
-      import_progress: 0,
-      taille_fichier: options.fileSize,
-      hash_secret: documentHash,
-      hash_fichier,
-      added_by,
-      updated_at: new Date(),
-      created_at: new Date(),
-    });
-  }
```
Comment (david-nathanael) [inline server/src/modules/actions/documents.actions.ts:347]: c'était jamais utilisé c'est ca ?

**Review by david-nathanael — APPROVED** [review]

File: server/src/modules/actions/documents.actions.ts (hunk 6 above)
Comment (moroine) [reply server/src/modules/actions/documents.actions.ts:347]: Nope, j'ai pas trouvé la string `createDocumentDb` dans le projet

**Pull request merged**